Packs/Salesforce/Integrations/SalesForceEventCollector/SalesForceEventCollector.py

```diff
@@ -217,53 +172,61 @@ def main():
 
-    request = Request(**demisto_params)
+    request = IntegrationHTTPRequest(**demisto_params)
 
+    # add the params to the url in order to make the request without decoding the params
     url = urljoin(demisto_params.get("url"), 'services/oauth2/token')
     request.url = f'{url}?grant_type=password&' \
                   f'client_id={demisto_params.get("client_id")}&' \
                   f'client_secret={demisto_params.get("client_secret")}&' \
                   f'username={demisto_params.get("username")}&' \
                   f'password={demisto_params.get("password")}'
 
-    client = Client(request)
+    options = IntegrationOptions.parse_obj(demisto_params)
+    client = SalesforceClient(request, options)
 
     after = get_timestamp_format(demisto_params.get('after'))
 
-    get_events = GetEvents(client, demisto_params.get('verify'),
-                           demisto_params.get('query'), after, demisto_params.get('last_id'))
+    get_events = SalesforceGetEvents(client, options, files_limit, demisto_params.get('query'),
+                                     after, demisto_params.get('last_id'))
 
     command = demisto.command()
     try:
-        urllib3.disable_warnings()
-
         if command == 'test-module':
-            get_events.aggregated_results(limit=1)
+            get_events.files_limit = 1
+            get_events.run()
             return_results('ok')
-        elif command == 'salesforce-get-events' or command == 'fetch-events':
-            events = get_events.aggregated_results(limit=int(demisto_params.get('limit')))
+        elif command in ('salesforce-get-events', 'fetch-events'):
+            events = get_events.run()
 
             if command == 'fetch-events':
-                while len(events) > 0:
-                    send_events_to_xsiam(events[:events_to_add_per_request], 'salesforce-audit', 'salesforce-audit')
-                    events = events[events_to_add_per_request:]
+                if events:
+                    demisto.setLastRun(get_events.get_last_run())
+                else:
+                    send_events_to_xsiam([], 'salesforce', demisto_params.get('product'))
+
             elif command == 'salesforce-get-events':
                 command_results = CommandResults(
-                    readable_output=tableToMarkdown('salesforce Logs', events, headerTransform=pascalToSpace),
+                    readable_output=tableToMarkdown('salesforce audit Logs', events, headerTransform=pascalToSpace),
                     outputs_prefix='salesforce.Logs',
                     outputs_key_field='timestamp',
                     outputs=events,
                     raw_response=events,
                 )
                 return_results(command_results)
+
+            while len(events) > 0:
+                send_events_to_xsiam(events[:events_to_add_per_request], 'salesforce', demisto_params.get('product'))
+                events = events[events_to_add_per_request:]
```

Comment:
There is no need to do it in batches.
A single call is enough.

Should only be called in fetch-events or when the relevant argument is given to the command.